### PR TITLE
Add exponential reconcile for serverless controller

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- customize exponential reconcile requeue
- apply exponential reconcile for git errors

**Related issue(s)**

See https://github.com/kyma-project/kyma/issues/12643

### CODE REVIEW

**Review by dbadura — COMMENTED** [review]
For those errors we shouldn't requeue function as those errors are not recoverable:
- ErrorCodeNotFound ErrorCode = C.GIT_ENOTFOUND
- ErrorCodeInvalidSpec ErrorCode = C.GIT_EINVALIDSPEC

File: components/function-controller/internal/controllers/serverless/function_reconcile.go [hunk 1]
```diff
@@ -58,6 +61,11 @@ func (r *FunctionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&corev1.Service{}).
 		Owns(&autoscalingv1.HorizontalPodAutoscaler{}).
 		WithOptions(controller.Options{
+			RateLimiter: workqueue.NewMaxOfRateLimiter(
```
Comment (dbadura) [inline components/function-controller/internal/controllers/serverless/function_reconcile.go:64]: Add unit test for it.

**Review by m00g3n — APPROVED** [review]
LGTM

**Pull request merged**